library/xml/src/modules/XML.rb

```diff
@@ -247,7 +247,11 @@ def parse_node(node, result)
       children = children.select(&:element?)
       # we need just direct text under node. Can be splitted with another elements
       # but remove whitespace only text
-      text = node.xpath("text()").text.strip
+      #
+      # Do NOT strip trailing white space in CDATA blocks. Maybe people put
+      # it intentionally there (bsc#1195910).
+      text_nodes = node.xpath("text()")
+      text = text_nodes.map { |n| n.cdata? ? n.content.lstrip : n.content.strip }.join
```

Comment:
is there reason for that lstrip in cdata? I expect it should be as it is

Reply:
You want to strip leading newlines in scripts.